core/errors.go

```diff
@@ -29,6 +30,8 @@ func isTemporary(info httperror.Info, err error) bool {
 		return true
 	case "CH001": // request timed out
 		return true
+	case "CH012": // conflict
+		return true
```

Comment:
Maybe we need two errors, the default of which is a permanent error. If the handler is doing a read-modify-write cycle, it can re-map the permanent error into a temporary error so that the client will retry.